core/src/View/Kokkos_ViewMapping.hpp

```diff
@@ -1754,6 +1762,43 @@ struct ViewOffset<
 
   /* Enable padding for trivial scalar types with non-zero trivial scalar size.
    */
+
+  template <unsigned TrivialScalarSize>
+  constexpr size_type compute_stride(const Kokkos::LayoutRight& arg_layout) {
+    return arg_layout.stride != KOKKOS_IMPL_CTOR_DEFAULT_ARG
+               ? arg_layout.stride
+               : Padding<TrivialScalarSize>::
+                     stride(/* 2 <= rank */
+                            m_dim.N1 *
+                            (dimension_type::rank == 2
+                                 ? size_t(1)
+                                 : m_dim.N2 *
+                                       (dimension_type::rank == 3
+                                            ? size_t(1)
+                                            : m_dim.N3 *
+                                                  (dimension_type::rank == 4
+                                                       ? size_t(1)
+                                                       : m_dim.N4 *
+                                                             (dimension_type::
+                                                                          rank ==
+                                                                      5
+                                                                  ? size_t(1)
+                                                                  : m_dim.N5 *
+                                                                        (dimension_type::
+                                                                                     rank ==
+                                                                                 6
+                                                                             ? size_t(
+                                                                                   1)
+                                                                             : m_dim.N6 *
+                                                                                   (dimension_type::
+                                                                                                rank ==
+                                                                                            7
+                                                                                        ? size_t(
+                                                                                              1)
+                                                                                        : m_dim
+                                                                                              .N7)))))));
```

Comment:
Since it's now in a fucntion, we don't have to write this as one statement.
```suggestion
    if (arg_layout.stride != KOKKOS_IMPL_CTOR_DEFAULT_ARG)
      return arg_layout.stride;
    size_type result = m_dim.N1;
    if (dimension_type::rank >= 3)
      result *= m_dim.N2;
    if (dimension_type::rank >= 4)
      result *= m_dim.N3;
    if (dimension_type::rank >= 5)
      result *= m_dim.N4;
    if (dimension_type::rank >= 6)
      result *= m_dim.N5;
    if (dimension_type::rank >= 7)
      result *= m_dim.N6;
    if (dimension_type::rank >= 8)
      result *= m_dim.N7;
    return Padding<TrivialScalarSize>::stride(result);
```